vizier/ui/src/info/vizierdb/ui/components/dataset/DatasetSummary.scala

```diff
@@ -19,14 +19,16 @@ import play.api.libs.json._
 class DatasetSummary(projectId: Identifier, datasetId: Identifier)(implicit val owner: Ctx.Owner) {
   
   implicit val ec: scala.concurrent.ExecutionContext = scala.concurrent.ExecutionContext.global
-  
-  def fetchAndRenderDatasetInfo(): Future[HtmlTag] = {
-    Vizier.api.artifactGet(
+
+  val artifact = Vizier.api.artifactGet(
       projectId,
       datasetId,
       limit = Some(0),
       profile = Some("true")
-    ).map {
+    )
+  
+  def fetchAndRenderDatasetInfo(): Future[HtmlTag] = {
+    artifact.map {
```

Comment:
I'm not sure I understand the purpose of this refactor... Is the idea to cache state?  If so, rename the function `fetchAndRender` is no longer appropriate.  In fact, if you're only going to call the artifact constructor once, you may as well render directly to an Rx.  e.g., see:

https://github.com/lihaoyi/scala.rx/tree/master?tab=readme-ov-file#future